plugins/camera/camera_impl.cpp

```diff
@@ -87,24 +92,76 @@ void CameraImpl::deinit()
         std::lock_guard<std::mutex> lock(_video_stream_info.mutex);
         _video_stream_info.callback = nullptr;
     }
+
+    _camera_found = false;
+}
+
+void CameraImpl::check_connection_status()
+{
+    // FIXME: This is a workaround because we don't want to be tied to the
+    // discovery of an autopilot which triggers enable() and disable() but
+    // we are interested if a camera is connected or not.
+    if (_parent->has_camera(_camera_id)) {
+        if (!_camera_found) {
+            _camera_found = true;
+            manual_enable();
+        }
+    }
 }
 
 void CameraImpl::enable()
+{
+    // FIXME: We check for the connection status manually because
+    // we're not interested int the connection state of the autopilot
+    // but only the camera.
+}
+
+void CameraImpl::manual_enable()
 {
     refresh_params();
-    request_flight_information();
 
+    LogDebug() << "request camera info";
     auto command_camera_info = make_command_request_camera_info();
     _parent->send_command_async(command_camera_info, nullptr);
 
+    request_flight_information();
+
     _parent->add_call_every(
         [this]() { request_flight_information(); }, 10.0, &_flight_information_call_every_cookie);
 }
 
 void CameraImpl::disable()
+{
+    // FIXME: We check for the connection status manually because
+    // we're not interested int the connection state of the autopilot
+    // but only the camera.
+}
+
+void CameraImpl::manual_disable()
 {
     invalidate_params();
     _parent->remove_call_every(_flight_information_call_every_cookie);
+
+    _camera_found = false;
+}
+
+Camera::Result CameraImpl::select_camera(unsigned id)
```

Comment:
Is the idea the following?

1. We instantiate a `camera` object
2. We select the desired camera
3. If needed, we can instantiate another `camera` object and set it to another camera

Reply:
I guess so. I agree it's ugly.